src/v/kafka/server/handlers/describe_configs.cc

```diff
@@ -77,12 +79,79 @@ static ss::sstring describe_as_string(const T& t) {
     return ssx::sformat("{}", t);
 }
 
+// Kafka protocol defines integral types by sizes. See
+// https://kafka.apache.org/protocol.html
+template<typename T>
+constexpr auto num_bits = CHAR_BIT * sizeof(T);
+
+template<typename T>
+constexpr bool is_short
+  = std::is_integral_v<T> && !std::is_same_v<T, bool> && num_bits<T> < 32;
+
+template<typename T>
+constexpr bool is_int = std::is_integral_v<T>&& num_bits<T> >= 32
+                        && num_bits<T> < 64;
+
+template<typename T>
+constexpr bool is_long = std::is_integral_v<T>&& num_bits<T> >= 64;
```

Comment:
nit: since the entire span of possible bit sizes is attempted to be covered, I think we should stay on the other side of the bit-number boundary of the types. `is_long` is essentially a check whether the value of the `T` would fit into `describe_configs_type::long_type` (please let me know if I misunderstand that). A hypotetical 72-bit integer won't, and therefore its `is_long` should be false. From that perspective, 
* `is_short` <= 16 bits,
* 16 bits < `is_int` <= 32 bits
* 32 bits < `is_long` <= 64 bits

However that's a nit because for 8/16/32/64 bits, the above logic works fine, it's just confusing.

Reply:
> nit: since the entire span of possible bit sizes is attempted to be covered, I think we should stay on the other side of the bit-number boundary of the types. `is_long` is essentially a check whether the value of the `T` would fit into `describe_configs_type::long_type` (please let me know if I misunderstand that).

I wouldn't say "fit" but yes. We are mapping the sizes of C++ types and some Redpanda objects to the Kafka API here.
https://kafka.apache.org/protocol.html
We are just using the type sizes for checking

Reply:
I think we will be OK with the bounds checking as it is.

Reply:
> https://kafka.apache.org/protocol.html
> We are just using the type sizes for checking

I can't see the sizes for SHORT/INT/LONG in that doc, they are only mentioned in DescribeConfig API and that's it. Am I missing something? Or maybe the author is referring to Java type sizes?

Reply:
> > https://kafka.apache.org/protocol.html
> > We are just using the type sizes for checking
> 
> I can't see the sizes for SHORT/INT/LONG in that doc, they are only mentioned in DescribeConfig API and that's it. Am I missing something? Or maybe the author is referring to Java type sizes?


Correct. I related Java type sizes to the Kafka protocol. I just updated the comment and the type checking.